packages/rainbowkit/src/components/ConnectOptions/MobileOptions.tsx

```diff
@@ -67,69 +68,62 @@ export function WalletButton({
-          // successful connection and then the universal provider uses it on requests. We call
-          // it upon onConnecting; this now needs to be called for both v1 and v2 Wagmi connectors.
-          // The `connector` type refers to Wagmi connectors, as opposed to RainbowKit wallet connectors.
-          // https://github.com/WalletConnect/web3modal/blob/27f2b1fa2509130c5548061816c42d4596156e81/packages/core/src/utils/CoreUtil.ts#L72
-          setWalletConnectDeepLink({ mobileUri, name });
-        }
+    const onMobileUri = async () => {
+      const mobileUri = await getMobileUri?.();
 
-        if (mobileUri.startsWith('http')) {
-          // Workaround for https://github.com/rainbow-me/rainbowkit/issues/524.
-          // Using 'window.open' causes issues on iOS in non-Safari browsers and
-          // WebViews where a blank tab is left behind after connecting.
-          // This is especially bad in some WebView scenarios (e.g. following a
-          // link from Twitter) where the user doesn't have any mechanism for
-          // closing the blank tab.
-          // For whatever reason, links with a target of "_blank" don't suffer
-          // from this problem, and programmatically clicking a detached link
-          // element with the same attributes also avoids the issue.
-          const link = document.createElement('a');
-          link.href = mobileUri;
-          link.target = '_blank';
-          link.rel = 'noreferrer noopener';
-          link.click();
-        } else {
-          window.location.href = mobileUri;
-        }
+      if (!mobileUri) return;
+
+      if (mobileUri) {
+        setWalletConnectDeepLink({ mobileUri, name });
+      }
+
+      if (mobileUri.startsWith('http')) {
+        // Workaround for https://github.com/rainbow-me/rainbowkit/issues/524.
+        // Using 'window.open' causes issues on iOS in non-Safari browsers and
+        // WebViews where a blank tab is left behind after connecting.
+        // This is especially bad in some WebView scenarios (e.g. following a
+        // link from Twitter) where the user doesn't have any mechanism for
+        // closing the blank tab.
+        // For whatever reason, links with a target of "_blank" don't suffer
+        // from this problem, and programmatically clicking a detached link
+        // element with the same attributes also avoids the issue.
+        const link = document.createElement('a');
+        link.href = mobileUri;
+        link.target = '_blank';
+        link.rel = 'noreferrer noopener';
+        link.click();
+      } else {
+        window.location.href = mobileUri;
       }
-    });
-  }, [connector, connect, getMobileUri, onConnecting, onClose, name, id]);
+    };
+
+    if (id !== wcId) onMobileUri();
+
+    // if the id is "walletConnect" then "showWalletConnectModal" will always be true
+    if (showWalletConnectModal) {
+      showWalletConnectModal();
+      onClose?.();
+      return;
+    }
```

Comment:
Good news. Fixed WalletConnect modal for mobile. Currently it's not working on mobile in `main` branch, but made sure to add it here 👌